examples/campaign_management/add_complete_campaigns_using_mutate_job.py

```diff
@@ -0,0 +1,536 @@
+#!/usr/bin/env python
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Adds complete campaigns using MutateJobService.
+
+Complete campaigns include campaign budgets, campaigns, ad groups and keywords.
+"""
+
+
+import argparse
+import sys
+from uuid import uuid4
+
+from google.ads.google_ads.client import GoogleAdsClient
+from google.ads.google_ads.errors import GoogleAdsException
+
+NUMBER_OF_CAMPAIGNS_TO_ADD = 2
+NUMBER_OF_AD_GROUPS_TO_ADD = 2
+NUMBER_OF_KEYWORDS_TO_ADD = 4
+MAX_TOTAL_POLL_INTERVAL_SECONDS = 60
+
+PAGE_SIZE = 1000
+
+
+def get_next_temporary_id():
+    """Returns the next temporary ID to use in mutate job operations.
+
+    If temporary ID has not been defined, i.e. if this function has not been
+    called before, we set the initial value as -1 and return it. Subsequent
+    calls will decrement the value by 1 before returning.
+
+    Returns: an int of the next temporary ID.
+    """
+    global temporary_id
```

Comment:
The purpose of the `global` keyword is to be used in functions to override the local scope, it doesn't make sense to use it at the module level outside of a function. 

I use it here so that I can define the variable when the function is first called. It's a little weird, but for the purposes of demonstration I want this variable to be set to -1 initially because that's the first ID that should be used, (as opposed to setting it to 0 then deprecating it before being returned for the first time). Basically I'm trying to mirror [this function](https://github.com/googleads/google-ads-php/blob/master/examples/CampaignManagement/AddCompleteCampaignsUsingMutateJob.php#L554) as defined in PHP.

I can refactor this so it doesn't use `global` but the logic may still be convoluted since we can't `return temporary_id -= 1`

Reply:
Alternative implementation looks like:

```python
_temporary_id = -1

def _get_next_temporary_id():                                                   

    global _temporary_id

    if _temporary_id == -1:
        return _temporary_id                                                    
    else:                                                                       
        _temporary_id -= 1                                                      
        return _temporary_id 
```

Is this preferable?

Reply:
or how about:
```
_temporary_id = 0

def _get_next_temporary_id():
    global _temporary_id
    _temporary_id -= 1
    return _temporary_id
```

Reply:
Yeah that works, but the point was to not set the var to 0 initially because it's not a valid temp ID. I'll use this anyway since it's less convoluted.